code/modules/mob/living/simple_animal/friendly/butterfly.dm

```diff
@@ -31,3 +31,148 @@
 
 /mob/living/simple_animal/butterfly/bee_friendly()
 	return TRUE //treaty signed at the Beeneeva convention
+
+/mob/living/simple_animal/fly
+	name = "swarm of flies"
+	desc = "[health = 1 ? "A single brave little fly, ready to take what the world throws at it." : "A swarm of flies, because the Janitor just isn't doing a good enough job."]"
+	icon_state = "fly-10"
+	icon_living = "fly-10"
+	icon_dead = "fly_dead"
+	turns_per_move = 1
+	response_help = "shoos"
+	response_disarm = "shoos"
+	response_harm = "splats"
+	speak_emote = list("buzzes")
+	maxHealth = 10 //it's a swarm!!
+	health = 10
+	harm_intent_damage = 3 //you can only kill one of the flies in the swarm at a time
+	friendly = "nudges"
+	density = FALSE
+	movement_type = FLYING
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	ventcrawler = VENTCRAWLER_ALWAYS
+	mob_size = MOB_SIZE_TINY
+	gold_core_spawnable = FRIENDLY_SPAWN
+	verb_say = "buzzes"
+	verb_ask = "buzzes inquisitively"
+	verb_exclaim = "buzzes intensely"
+	verb_yell = "buzzes intensely"
+
+/mob/living/simple_animal/fly/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	src.health --
+	visible_message("the fly stops moving...")
+	if(stat)
+		icon_state = "fly-[health]"
+		maxHealth = health
+
+/mob/living/simple_animal/fly/time
+	name = "swarm of time flies"
+	desc = "Radiation seems to have given this [health = 1 ? "fly" : "swarm of flies"] time bending powers."
+	icon_state = "timefly-10"
+	icon_living = "timefly-10"
+	icon_dead = "timefly_dead"
+
+/mob/living/simple_animal/fly/time/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	src.health --
+	visible_message("the time fly stops moving...")
+	if(stat)
+		icon_state = "timefly-[health]"
+		maxHealth = health
+
+/mob/living/simple_animal/fly/time/AttackingTarget()
+	. = ..()
+	if(. && isliving(target)
+	var/mob/living/L = target
+
+
+
+
+
+/obj/effect/proc_holder/spell/targeted/timewarp
+	name = "Time Warp!"
+	desc = "Warps you 10 seconds into the future! If, for some reason, you wanted to teleport yourself 10 seconds into the future."
+	charge_max = 0
+	clothes_req = 0
+	phase_allowed = 1
+	selection_type = "range"
+	range = -1
+	include_user = 1
+	cooldown_min = 0
+	overlay = null
+	action_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	action_icon_state = "ninja_cloak"
+	action_background_icon_state = "bg_alien"
+
+/obj/effect/proc_holder/spell/targeted/timewarp/cast(list/targets,mob/living/user = usr)
+	playsound(get_turf(user), 'sound/magic/ethereal_enter.ogg', 50, 1, -1)
+	visible_message("<span class='boldwarning'>[user] melts into the shadows!</span>")
+	user.SetStun(0, FALSE)
+	user.SetKnockdown(0, FALSE)
+	user.setStaminaLoss(0, 0)
+	var/obj/effect/dummy/shadow/S2 = new(get_turf(user.loc))
+	user.forceMove(S2)
+	S2.jaunter = user
+	else
+		to_chat(user, "<span class='warning'>It isn't dark enough here!</span>")
```

Comment:
This seems like a remnant from copy/pasting this from somewhere else, since there's no IF before it. Should probably get rid of it.

Reply:
It's my grue code please hold

Reply:
is your grue code also stolen from nightmares lol?